src/vellum/workflows/runner/runner.py

```diff
@@ -565,6 +572,16 @@ def initiate_node_streaming_output(
 
                 with execution_context(parent_context=updated_parent_context, trace_id=execution.trace_id):
                     for output in node_run_response:
+                        try:
+                            default_serializer(output)
+                        except (TypeError, ValueError) as exc:
+                            raise NodeException(
+                                message=(
+                                    f"Node {node.__class__.__name__} produced output: "
+                                    f"'{output.name}' that could not be serialized to JSON: {exc}"
+                                ),
+                                code=WorkflowErrorCode.INVALID_OUTPUTS,
+                            ) from exc
```

Comment:
[blocking] I would rather support than add this overhead on every chunk from a node. Should be as easy as updating `VellumJsonEncoder` I believe